client/content/markdown.test.ts

```diff
@@ -0,0 +1,115 @@
+// @vitest-environment nuxt
+import fsPromises from 'node:fs/promises'
+import path from 'node:path'
+import { test, expect } from 'vitest'
+import { micromark } from 'micromark'
+import { globby } from 'globby'
+import { parseHtml } from '~/utilities/html'
+
+const __dirname = import.meta.dirname
+const clientPath = path.resolve(__dirname, '..')
+const contentPath = path.resolve(clientPath, 'content')
+
+test('Markdown links validation', async () => {
+  // Unfortunately Nuxt Content's queryCollection can't run in tests only in server routes
+  const markdownPaths = await globby('**/*.md', {
+    cwd: contentPath
+  })
+  const markdowns = await Promise.all(
+    markdownPaths.map((markdownPath) =>
+      fsPromises.readFile(path.join(contentPath, markdownPath), 'utf-8')
+    )
+  )
+  const htmls = markdowns.map((markdown) => micromark(markdown))
+
+  const validateLink = (href: string, markdownPath: string): void => {
+    if (
+      href.startsWith('//') ||
+      href.startsWith('http://') ||
+      href.startsWith('https://') ||
+      href.startsWith('mailto:')
+    ) {
+      // external links aren't validated
+      return
+    }
+
+    const pathname = new URL(href, 'http://localhost/').pathname // removes #hash link
+
+    /**
+     * In order to ensure validate links (ie trailing slashes in most cases)
+     * we want to generate
```

Comment:
Having trouble parsing this comment...

Reply:
Good spotting! I'll fix that in the next PR.